aptos-move/framework/aptos-framework/sources/multisig_account.move

```diff
@@ -0,0 +1,1477 @@
+    struct TransactionExecutionSucceededEvent has drop, store {
+        executor: address,
+        transaction_id: u64,
+        transaction_payload: vector<u8>,
+        num_approvals: u64,
+    }
+
+    /// Event emitted when a transaction's execution failed.
+    struct TransactionExecutionFailedEvent has drop, store {
+        executor: address,
+        transaction_id: u64,
+        transaction_payload: vector<u8>,
+        num_approvals: u64,
+        execution_error: ExecutionError,
+    }
+
+    #[view]
+    /// Return the number of signatures required to execute or execute-reject a transaction in the provided
+    /// multisig account.
+    public fun signatures_required(multisig_account: address): u64 acquires MultisigAccount {
+        borrow_global<MultisigAccount>(multisig_account).signatures_required
+    }
+
+    #[view]
+    /// Return a vector of all of the provided multisig account's owners.
+    public fun owners(multisig_account: address): vector<address> acquires MultisigAccount {
+        borrow_global<MultisigAccount>(multisig_account).owners
+    }
+
+    #[view]
+    /// Return the transaction with the given transaction id.
+    public fun get_transaction(
+        multisig_account: address,
+        transaction_id: u64,
+    ): MultisigTransaction acquires MultisigAccount {
+        *table::borrow(&borrow_global<MultisigAccount>(multisig_account).transactions, transaction_id)
+    }
+
+    #[view]
+    /// Return the payload for the next transaction in the queue.
+    public fun get_next_transaction_payload(
+        multisig_account: address, provided_payload: vector<u8>): vector<u8> acquires MultisigAccount {
+        let multisig_account_resource = borrow_global<MultisigAccount>(multisig_account);
+        let transaction_id = multisig_account_resource.last_executed_transaction_id + 1;
+        let transaction = table::borrow(&multisig_account_resource.transactions, transaction_id);
+
+        if (option::is_some(&transaction.payload)) {
+            *option::borrow(&transaction.payload)
+        } else {
+            provided_payload
+        }
+    }
+
+    #[view]
+    /// Return true if the transaction with given transaction id can be executed now.
+    public fun can_be_executed(
+        multisig_account: address, transaction_id: u64): bool acquires MultisigAccount {
+        let multisig_account_resource = borrow_global_mut<MultisigAccount>(multisig_account);
+        let transaction = table::borrow(&mut multisig_account_resource.transactions, transaction_id);
+        let (num_approvals, _) = num_approvals_and_rejections(&multisig_account_resource.owners, transaction);
+        transaction_id == multisig_account_resource.last_executed_transaction_id + 1 &&
+            num_approvals >= multisig_account_resource.signatures_required
+    }
+
+    #[view]
+    /// Return true if the transaction with given transaction id can be officially rejected.
+    public fun can_be_execute_rejected(
```

Comment:
the name is a bit weird, should it just be `can_be_rejected`?

Reply:
This is because reject = an owner voting no on a transaction, and execute_rejected officially removes a tx after there are enough rejections. But for can_be_execute_rejected, it does sound strange. Shortening it to can_be_rejected